### PR TITLE
New Lint quick-fix: avoid_init_to_null

### PR DESCRIPTION
**Pull request by JPaulsen**
From [Linter](http://dart-lang.github.io/linter/lints/avoid_init_to_null.html)

### CODE REVIEW

File: pkg/analysis_server/lib/src/services/correction/fix.dart [hunk 1]
```diff
@@ -185,6 +186,8 @@ class DartFixKind {
       const FixKind('INSERT_SEMICOLON', 50, "Insert ';'");
   static const LINT_ADD_OVERRIDE =
       const FixKind('LINT_ADD_OVERRIDE', 50, "Add '@override' annotation");
+  static const LINT_REMOVE_INIT_TO_NULL =
+      const FixKind('LINT_REMOVE_INIT_TO_NULL', 50, "Remove initialization to null");
```
Comment (bwilkerson) [inline pkg/analysis_server/lib/src/services/correction/fix.dart]: Please rename to `REMOVE_INITIALIZER` and change the description appropriately. There's nothing about this (or any other) fix that is specific to a lint, and the fix isn't dependent on the form of the initialization expression.

File: pkg/analysis_server/lib/src/services/correction/fix_internal.dart [hunk 2]
```diff
@@ -2330,6 +2333,20 @@ class FixProcessor {
     _addFix(DartFixKind.LINT_ADD_OVERRIDE, []);
   }
 
+  void _addLintFixRemoveNullInit() {
```
Comment (bwilkerson) [inline pkg/analysis_server/lib/src/services/correction/fix_internal.dart]: Please rename to `_addFix_removeInitializer`. First, it corresponds to the naming convention of the majority of the methods in this class, and second it more accurately represents the fact that the method doesn't care what the initializer looks like, it just removes it.

File: pkg/analysis_server/test/services/correction/fix_test.dart [hunk 3]
```diff
@@ -5897,6 +5897,36 @@ class Sub extends Test {
 ''');
   }
 
+  test_lint_avoid_init_to_null_top_level() async {
```
Comment (bwilkerson) [inline pkg/analysis_server/test/services/correction/fix_test.dart]: Please rename to `test_removeInitializer_topLevel`. Although it doesn't exist yet, I'd like to establish a pattern of `test_<fix name as camel case>_<qualifier>`.

File: pkg/analysis_server/test/services/correction/fix_test.dart [hunk 4]
```diff
@@ -5897,6 +5897,36 @@ class Sub extends Test {
...
+    await findLint(src, LintNames.avoid_init_to_null);
+
+    await applyFix(DartFixKind.LINT_REMOVE_INIT_TO_NULL);
+
+    verifyResult('''
+var x;
+''');
+  }
+
+  test_lint_avoid_init_to_null_field() async {
```
Comment (bwilkerson) [inline pkg/analysis_server/test/services/correction/fix_test.dart]: Similarly, please rename to `test_removeInitializer_field`.

File: pkg/analysis_server/test/services/correction/fix_test.dart [hunk 5]
```diff
@@ -5897,6 +5897,36 @@ class Sub extends Test {
...
+
+    await applyFix(DartFixKind.LINT_REMOVE_INIT_TO_NULL);
+
+    verifyResult('''
+class Test {
+  int x;
+}
+''');
+  }
+
```
Comment (bwilkerson) [inline pkg/analysis_server/test/services/correction/fix_test.dart]: We should have at least one test where the initializer is in the middle of a list of variable declarations. Perhaps something like:
```
void f() {
  String a = 'a', b = null, c = 'c';
}
```
(which will also test local variables, covering another missing case).

File: pkg/analysis_server/lib/src/services/correction/fix.dart (hunk 1 above)
Comment (JPaulsen) [inline pkg/analysis_server/lib/src/services/correction/fix.dart]: So REMOVE_INITIALIZER or LINT_REMOVE_INITIALIZER? We tried to do just like the other 2 linter quick fixes (LINT_ADD_OVERRIDE, LINT_REMOVE_INTERPOLATION_BRACES

File: pkg/analysis_server/lib/src/services/correction/fix_internal.dart (hunk 2 above)
Comment (JPaulsen) [inline pkg/analysis_server/lib/src/services/correction/fix_internal.dart]: Fixed

File: pkg/analysis_server/test/services/correction/fix_test.dart (hunk 3 above)
Comment (JPaulsen) [inline pkg/analysis_server/test/services/correction/fix_test.dart]: Fixed

File: pkg/analysis_server/test/services/correction/fix_test.dart (hunk 4 above)
Comment (JPaulsen) [inline pkg/analysis_server/test/services/correction/fix_test.dart]: Fixed

File: pkg/analysis_server/test/services/correction/fix_test.dart (hunk 5 above)
Comment (JPaulsen) [inline pkg/analysis_server/test/services/correction/fix_test.dart]: Fixed

**Review by bwilkerson — APPROVED** [review]

**Pull request merged**